landoapi/api/try_push.py

```diff
@@ -0,0 +1,222 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import base64
+import io
+import logging
+import math
+
+from connexion import ProblemException
+from dateutil.parser import parse as dateparse
+from flask import (
+    current_app,
+    g,
+)
+
+from landoapi import auth
+from landoapi.hgexports import (
+    GitPatchHelper,
+    HgPatchHelper,
+)
+from landoapi.models.landing_job import (
+    LandingJobStatus,
+    add_job_with_revisions,
+)
+from landoapi.models.revisions import Revision
+from landoapi.repos import (
+    SCM_LEVEL_1,
+    get_repos_for_env,
+)
+
+logger = logging.getLogger(__name__)
+
+
+# Borrowed from Mercurial core.
```

Comment:
Can this not be imported directly from the `mercurial` module?

```python
from mercurial.utils import stringutil
```

Same applies for anything else copy/pasted.

Reply:
When you sent this it occurred to me that I can't use this code due to incompatible licenses. I found a different solution that uses functions in the standard library.